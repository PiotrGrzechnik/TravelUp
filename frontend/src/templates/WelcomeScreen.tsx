import React from 'react'
import styled from 'styled-components'

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 100%;
  height: 100vh;
  min-height: 630px;
  background-color: ${({ theme }) => theme.colors.primaryAll[1]};
  background: linear-gradient(
      ${({ mask }) => (mask ? mask : 'rgba(0, 0, 0, 0), rgba(0, 0, 0, 0)')}
    )
    ${({ background }) => background && `, url(${background})`};
  background-size: cover;
`

type WelcomeScreenProps = {
  children: React.ReactNode
  background?: string
  mask?: string
}

const WelcomeScreen = (props: WelcomeScreenProps) => {
  const { children } = props

  return <Container {...props}>{children}</Container>
}

export default WelcomeScreen
