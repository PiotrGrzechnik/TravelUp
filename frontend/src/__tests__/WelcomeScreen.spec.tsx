import React from 'react'
import Enzyme, { shallow, mount, render } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { ThemeProvider } from 'styled-components'
import 'jest-styled-components'

import WelcomeScreen from 'src/containers/WelcomeScreen'

Enzyme.configure({ adapter: new Adapter() })

describe('WelcomeScreen', () => {
  test('renders component without crashing', () => {
    const wrapper = shallow(
      <WelcomeScreen>
        <h2>Hello!</h2>
      </WelcomeScreen>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
