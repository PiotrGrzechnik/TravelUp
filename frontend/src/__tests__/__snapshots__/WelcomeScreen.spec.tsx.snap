// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen renders component without crashing 1`] = `
<styled.div>
  <h2>
    Hello!
  </h2>
</styled.div>
`;
